src/library_wasmfs.js

```diff
@@ -187,6 +191,69 @@ var WasmFSLibrary = {
 #endif
     return wasmFS$backends[backend].getSize(file);
   },
+
+  // ProxiedAsyncJSImpl. Each function receives a function pointer and a
+  // parameter. We convert those into a convenient Promise API for the
+  // implementors of backends: the hooks we call should return Promises, which
+  // we then connect to the calling C++.
+
+  _wasmfs_jsimpl_async_alloc_file: async function(backend, file, fptr, arg) {
+#if ASSERTIONS
+    assert(wasmFS$backends[backend]);
+#endif
+    {{{ runtimeKeepalivePush() }}}
```

Comment:
Why is this pushing and popping necessary? Is there a way to make it less manual, like a `withRuntimeKeptAlive(...)` wrapper function?

Reply:
@sbc100 , did we consider a wrapper function? It's not less code, and only works in some situations, but might be nice.

Another option, for places where we can use `await`, is to have a macro `{{{ makeAwait }}}` perhaps that would put the push/pop around it?